core/src/validator.rs

```diff
@@ -168,7 +170,44 @@ impl Validator {
             config.storage_slots_per_turn,
             bank.slots_per_segment(),
         );
-
+        let bf = bank_forks.clone();
+        let kp = keypair.clone();
+        let mut counter = 0;
+        let counter_ref = &mut counter;
+        if counter == 0 {
+            set_handler(move || { 
```

Comment:
I just realized we'll want to add a config item in `ValidatorConfig` to enable this feature (disabled by default).  Then enable it explicitly in `validator/src/main.rs`.   We need this because there are some tests (the local_cluster/ tests) that create multiple Validators in the same process, and they all can't set a Ctrl+C handler

Reply:
could you give me more context / any code example how to do it?

Reply:
Around this area of the code, https://github.com/solana-labs/solana/blob/62c22c6cb19931d228986346fdf2b98fe5bfa27f/validator/src/lib.rs#L432-L437, add a `validator_config.enable_ctrl_c_handler = true` or so.   

Then in https://github.com/solana-labs/solana/blob/d379786c902d319ee739b1c88da822cec9b780b4/core/src/validator.rs#L57-L71, default `enable_ctrl_c_handler` to `false` and that'll take care of the local_cluster use case I mentioned above